File: Find Largest Value in Each Tree Row/FindLargestValueinEachTreeRow.java
import java.util.*;

public class FindLargestValueinEachTreeRow {
    public List<Integer> largestValues(TreeNode root) {
        Map<Integer,Integer> currentLargestValueMap = new HashMap<Integer,Integer>();
        DFS(root, 1, currentLargestValueMap);
        List<Integer> ret = new ArrayList<Integer>(currentLargestValueMap.size());
        for(int i=1;i<=currentLargestValueMap.size();i++){
        	ret.add(currentLargestValueMap.get(i));
        }
        
        return ret;
    }
    
    private void DFS(TreeNode node, int level, Map<Integer,Integer> largestValueMap){
    	if(node == null) return;
    	if(largestValueMap.containsKey(level)){
    		largestValueMap.put(level, Math.max(node.val, largestValueMap.get(level)));
    	}else{
    		largestValueMap.put(level, node.val);
    	}
    	
    	DFS(node.left, level+1, largestValueMap);
    	DFS(node.right, level+1, largestValueMap);
    }
}

File: Find Largest Value in Each Tree Row/FindLargestValueinEachTreeRow.cs
using System;
using System.Collections.Generic;

namespace LeetcodePracticeCsharpVersion
{
    class FindLargestValueinEachTreeRow
    {
        public IList<int> LargestValues(TreeNode root)
        {
            var currentLargestValueDict = new Dictionary<int, int>();
            DFS(root, 1, currentLargestValueDict);

            var ret = new List<int>(currentLargestValueDict.Count);

            for(int i = 0; i < currentLargestValueDict.Count; i++)
            {
                ret.Add(currentLargestValueDict[i + 1]);
            }

            return ret;
        }

        private void DFS(TreeNode node, int level, IDictionary<int, int> largestValueDict)
        {
            if (node == null) return;
            var value = node.val;
            if (largestValueDict.ContainsKey(level))
            {
                largestValueDict[level] = Math.Max(value, largestValueDict[level]);
            }
            else
            {
                largestValueDict.Add(level, value);
            }

            DFS(node.left, level + 1, largestValueDict);
            DFS(node.right, level + 1, largestValueDict);
        }
    }
}
